packages/icons/src/loaders/CharcoalIconFilesLoader.ts

```diff
@@ -0,0 +1,44 @@
+import { PixivIconLoadError } from './PixivIconLoadError'
+import { Loadable } from './Loadable'
+import charcoalIconFiles, { KnownIconFile } from '../charcoalIconFiles'
+
+/**
+ * `@charcoal-ui/icon-files` に収録されているアイコンを取ってくる
+ */
+export class CharcoalIconFilesLoader implements Loadable {
+  private _name: KnownIconFile
+  private _resultSvg: string | undefined = undefined
+  private _promise: Promise<string> | undefined = undefined
+
+  constructor(name: KnownIconFile) {
+    this._name = name
+  }
+
+  get importIconFile() {
+    return charcoalIconFiles[this._name]
+  }
+
+  async fetch(): Promise<string> {
+    if (this._resultSvg !== undefined) {
+      return this._resultSvg
+    }
+
+    if (this._promise) {
+      return this._promise
+    }
+
+    this._promise = this.importIconFile()
+      .then((svg) => {
+        this._resultSvg = svg
+        return this._resultSvg
+      })
+      .catch(() => {
+        throw new PixivIconLoadError(this._name)
```

Comment:
デバッグのしやすさを考えると`importIconFile`が投げたエラーが具体的に何だったのかわかるようにしておくのが良さそうです。

単にコンソールに出力するか、

```suggestion
      .catch((e) => {
        console.error(e)
        throw new PixivIconLoadError(this._name)
```

あるいは`PixivIconLoadError`のコンストラクタに`Error`を渡せるようにして、具体的なエラーの情報をmessageに含めるのはどうでしょうか

```ts
export class PixivIconLoadError extends Error {
  constructor(name: string, causedBy?: { toString(): string }) {
    super(`Failed to fetch <pixiv-icon name="${name}">${causedBy === undefined ? '' : `: ${causedBy}`}`)
    Object.setPrototypeOf(this, new.target)
  }
}
```

```suggestion
      .catch((e) => {
        throw new PixivIconLoadError(this._name, e)
```

Reply:
ES2022 だと `Error.prototype.cause` があるので理想的にはそっちに乗るほうが良さそうなんですよね。

https://zenn.dev/petamoriken/articles/bb123b2f50cdab

polyfill 入れないとつらい感じになるとかなら挙げてもらった方法にしてもいいですが